### ISSUE TITLE
Investigate cache job runtime increase, part 2!

### ISSUE THREAD

**Issue by hschallhorn** [department-of-veterans-affairs/caseflow#13384]
Runtimes for [the `UpdateCachedAppealsAttributesJob`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/jobs/update_cached_appeals_attributes_job.rb) increased from significantly after [recent changes](https://github.com/department-of-veterans-affairs/caseflow/pull/13285). Short runtimes allow us to run the job more frequently and keep the cache more up-to-date. This ticket exists to determine why runtimes increased and address it if possible. 

![Screen Shot 2020-02-11 at 12 23 42 PM](https://user-images.githubusercontent.com/45575454/74261583-7d25fb00-4cc9-11ea-9a9d-12bd6d7e6e73.png)

_[Source: Datadog](https://app.datadoghq.com/dashboard/t5t-exj-m32/cronjob-statistics)_

## Acceptance criteria
* [ ] Determine the cause of the increased runtime length
* [ ] Determine if the runtime length can be easily optimized to reduce runtime
* [ ] Make the optimization

**Comment by hschallhorn** [comment]
Appears to be a pretty clear case of an introduction of a few N+1 queries

**Comment by pkarman** [comment]
There are multiple reasons for this:

1. suggested_hearing_location triggers multiple SQL calls. Adding `.includes(:available_hearing_locations)` prefetches those.

2. The `power_of_attorney_name` is very expensive. For AMA appeals that is coming from BGS. We were already caching some BGS POA info, but only for 24 hours, and only based on upcoming hearings. We optimized this in a few ways:

 * increase the cache TTL to 30.days and prefetch oldest claimants for open Appeals each night, setting `updated_at` on the claimant for each one (reduce calls to BGS)
 * prefetch all the claimants for AMA appeals so we don't make so many db calls

For Legacy appeals this is more tricky. We can't easily prefetch claimants because `LegacyAppealRepresentative` has a lot of logic in it, and for VACOLS calls, we must make at least 2 SQL statements for each case. We were able to reduce the VACOLS calls from 5 to 2, so that's a big help, but pre-fetching representatives would be a bigger help.
